File: backend/services/core-service/src/main/java/kr/lastdish/core/order/presentation/dto/OrderRejectReason.java
package kr.lastdish.core.order.presentation.dto;

public enum OrderRejectReason {
  OUT_OF_STOCK(false, "재고 소진으로 주문이 취소되었어요. 결제하신 금액은 예치금으로 환불됩니다."),
  QUALITY_ISSUE(false, "상품 준비 중 문제가 발견되어 주문이 취소되었어요. 결제하신 금액은 예치금으로 환불됩니다."),
  NOT_READY(true, "매장 사정으로 준비가 어려워 주문이 취소되었어요. 결제하신 금액은 예치금으로 환불됩니다."),
  STORE_CLOSED(false, "매장 사정으로 오늘 픽업이 어려워 주문이 취소되었어요. 결제하신 금액은 예치금으로 환불됩니다."),
  SYSTEM_ERROR(true, "일시적인 오류로 주문이 취소되었어요. 결제하신 금액은 예치금으로 환불됩니다. 불편을 드려 죄송합니다.");

  private final boolean restoreStock;
  private final String message;

  OrderRejectReason(boolean restoreStock, String message) {
    this.restoreStock = restoreStock;
    this.message = message;
  }

  public boolean shouldRestoreStock() {
    return restoreStock;
  }

  public String getMessage() {
    return message;
  }
}

File: frontend/lib/domain/model/order_reject_reason.dart
/// 매장 거절 사유(백엔드 `OrderRejectReason` enum, ADR 012로 확정). order_status.dart와
/// 같은 패턴 — 서버는 영문 enum명 그대로 주고받고, 한글 라벨은 프론트에서 매핑한다.
/// 구매자 취소(`cancelReason`, 자유 텍스트)와는 무관 — 이건 매장 거절(S3) 전용이다.
const List<String> orderRejectReasonValues = [
  'OUT_OF_STOCK',
  'QUALITY_ISSUE',
  'NOT_READY',
  'STORE_CLOSED',
  'SYSTEM_ERROR',
];

const Map<String, String> orderRejectReasonDisplayNames = {
  'OUT_OF_STOCK': '재고 소진',
  'QUALITY_ISSUE': '품질 문제',
  'NOT_READY': '준비 지연',
  'STORE_CLOSED': '매장 휴무',
  'SYSTEM_ERROR': '시스템 오류',
};

String orderRejectReasonLabel(String value) =>
    orderRejectReasonDisplayNames[value] ?? value;
